### ISSUE TITLE
error updating omf

### ISSUE THREAD

**Issue by ninjatux** [oh-my-fish/oh-my-fish#290]
Hi guys, I'm having problems running `omf update`, this the console output:

```bash
~/workspace $ omf update
Updating Oh My Fish...
Oh My Fish is up to date.
✔ brew successfully updated.
✔ osx successfully updated.
✔ python successfully updated.
✔ tab successfully updated.
fish: Unknown command 'refresh'
~/.local/share/omf/pkg/omf/omf.fish (line 118):       refresh
                                                      ^
in function 'omf'
	called on standard input
	with parameter list 'update'
```

and if I try to print out the version:

```bash
~/workspace $ omf --version
fish: Unknown command 'omf.version'
~/.local/share/omf/pkg/omf/omf.fish (line 11):       omf.version
                                                     ^
in function 'omf'
	called on standard input
	with parameter list '--version'
```

some extra infos:

```
System Version: OS X 10.11.3 (15D21)
Kernel Version: Darwin 15.3.0
Python: 2.7.10
Homebrew 0.9.5 (git revision 6c12b; last commit 2016-03-11)
```

**Comment by oranja** [comment]
Hi, we'll need a little bit more information if we hope to get this right the first time.
* When and how did you install OMF?
* When was the last time you updated OMF?
```git -C $OMF_PATH log -1```
* What is your fish version?
```fish --version```

**Comment by ninjatux** [comment]
sure @oranja 

* I installed it following the instructions on the readme, `curl -L https://github.com/oh-my-fish/oh-my-fish/raw/master/bin/install | fish` and was beginning of september, checking some logs looks like the 8 of september
* last update
```
~/workspace $ git -C $OMF_PATH log -1

commit 5260fd496afbd60a317ae41ecdd739d364e0580c
Merge: 0699c9f 0fc99b3
Author: Justin Hileman <github@0x7f.us>
Date:   Mon Mar 28 07:21:56 2016 -0700

    Merge pull request #287 from oranja/pkg-uptodate

    bugfix: Core update reported failure when up-to-date
```
* my fish version is 2.2.0

**Comment by arashm** [comment]
I had the same problem I updated the git repo and it's fixed now:

```bash
cd $OMF_PATH
git pull --rebase
```

**Comment by oranja** [comment]
Hey, sorry I didn't get back to this issue earlier.

What I believe that have you both have in common is that you haven't updated OMF for quite a while, until recently.

_Background_: There was a big change, in which most of the core component was reorganized internally. Among these changes, the call to `refresh` was removed from `omf update` and a lot of functions moved places. We made an effort to preserve compatibility and not break the update procedure, but it was based on what we had and knew back then. 

`refresh` was deprecated recently. Its place was taken by a safer `reload` routine. The `refresh` function is still there, but two things happened that made this bug surface anyway:
(A) In our tests there were no problems. It probably didn't occur to us that there can be a problem, because we update OMF regularly in the course of our work. Thus, there was no call to `refresh` during `omf update`, and even if there were, it would be recognized by _fish_ in its contemporary location.
(B) In your case, because you skipped some big changes, the `refresh` function's path was not in `fish_function_path` (_omf_ didn't tell _fish_ where to find it). And, of course, `omf update` was still looking for `refresh`, as we've seen in your logs of the failure.

_For you_: No damage was done by this error. Your OMF and packages got all the updates. To make sure that everything is in sync, simply run `omf reload` in present instances of fish. In new instances of fish there won't be a problem. (Correct me if I'm wrong).

_For us_: We can try to be more careful in the future, but there is always a limit of how far back we can go and still keep the system, and its development pace, sane. Soon we will begin to tag version numbers and therefore it will be easier to announce when it's a good time to `omf update`.

p.s. It was silly of me to ask ninjatux for the last update time -- after the update.. What I actually intended to know was the time of the second latest update. Trouble is, it's hard to get this information in retrospect.
